sql/schema/timescale.sql

```diff
@@ -47,31 +140,55 @@ CREATE TABLE v2_task_lookup_table (
     PRIMARY KEY (external_id)
 );
 
-CREATE OR REPLACE FUNCTION v2_tasks_olap_lookup_table_insert_function()
+CREATE OR REPLACE FUNCTION v2_tasks_olap_insert_function()
 RETURNS TRIGGER AS
 $$
 BEGIN
+    INSERT INTO v2_task_statuses_olap (id, inserted_at, tenant_id, workflow_id, readable_status)
+    SELECT id, inserted_at, tenant_id, workflow_id, readable_status
+    FROM new_rows;
+
     INSERT INTO v2_task_lookup_table (
         tenant_id,
         external_id,
         task_id,
         inserted_at
     )
-    VALUES (
-        NEW.tenant_id,
-        NEW.external_id,
-        NEW.id,
-        NEW.inserted_at
-    );
-    RETURN NEW;
+    SELECT tenant_id, external_id, id, inserted_at
+    FROM new_rows;
+    
+    RETURN NULL;
 END;
 $$
 LANGUAGE plpgsql;
 
-CREATE TRIGGER v2_tasks_olap_lookup_table_insert_trigger
+CREATE TRIGGER v2_tasks_olap_status_insert_trigger
 AFTER INSERT ON v2_tasks_olap
-FOR EACH ROW
-EXECUTE PROCEDURE v2_tasks_olap_lookup_table_insert_function();
+REFERENCING NEW TABLE AS new_rows
+FOR EACH STATEMENT
+EXECUTE FUNCTION v2_tasks_olap_insert_function();
+
+CREATE OR REPLACE FUNCTION v2_tasks_olap_status_update_function()
+RETURNS TRIGGER AS
+$$
+BEGIN
+    UPDATE v2_task_statuses_olap t
+    SET tenant_id = n.tenant_id,
+        workflow_id = n.workflow_id,
+        readable_status = n.readable_status
+    FROM new_rows n
+    WHERE t.id = n.id
+      AND t.inserted_at = n.inserted_at;
```

Comment:
qq - do we need this second condition?

Reply:
the primary key technically has it and the `id` isn't an identity column, it's a bigint which is sourced from the engine, so it might be necessary